src/librustc_borrowck/borrowck/mod.rs

```diff
@@ -913,6 +927,43 @@ impl<'a, 'tcx> BorrowckCtxt<'a, 'tcx> {
         err.emit();
     }
 
+    fn immutable_argument_should_be_mut(&self, nid: ast::NodeId, db: &mut DiagnosticBuilder) {
+        let parent = self.tcx.map.get_parent_node(nid);
+
+        if let Some(fn_like) = FnLikeNode::from_node(self.tcx.map.get(parent)) {
```

Comment:
So...this fn is pretty hard to read. =) Maybe we can use some temporaries or something? I just find it hard to follow, but maybe it's ok, if there was at least a comment. Basically when I read it I just see "walk a bunch of data structures to pull out data", but my eyes slide over the details...